src/main/kotlin/com/getkeepsafe/dexcount/SourceFile.kt

```diff
@@ -111,11 +111,27 @@ internal class DexFile(
         @JvmStatic
         private fun extractDexFromAar(file: File, dxTimeoutSecs: Int): List<DexFile> {
             // unzip classes.jar from the AAR
+            var minSdk = 13 // the default minSdkVersion of dx
             val tempClasses = file.unzip { entries ->
-                val jarFile = entries.find { it.name.matches(Regex("classes\\.jar")) }
-                makeTempFile("classes.jar").also {
-                    jarFile!!.writeTo(it)
+                var tempFile: File? = null
+
+                for (entry in entries) {
+                    if (entry.name == "AndroidManifest.xml") {
+                        val manifestText = entry.inputStream().bufferedReader().use { it.readText() }
+                        val match = Regex("""android:minSdkVersion="(\d+)"""").find(manifestText)
+                        if (match != null) {
+                            minSdk = match.groupValues[1].toInt()
+                        }
+                    }
+
+                    if (entry.name.matches(Regex("classes\\.jar"))) {
+                        tempFile = makeTempFile("classes.jar").also {
+                            entry.writeTo(it)
+                        }
+                    }
```

Comment:
Neither yours nor mine works if a variable is used for the min sdk, but this is a step closer:
```suggestion
                    when {
                        entry.name == "AndroidManifest.xml" -> {
                            val manifestText = entry.inputStream().bufferedReader().use { it.readText() }
                            val match = Regex("""android:minSdkVersion="(\d+)"""").find(manifestText)
                            if (match != null) {
                                minSdk = match.groupValues[1].toInt()
                            }
                        }
                        entry.name == "build.gradle" -> {
                            val gradleText = entry.inputStream().bufferedReader().use { it.readText() }
                            val match = Regex("""minSdkVersion (\d+)""").find(gradleText)
                            if (match != null) {
                                minSdk = match.groupValues[1].toInt()
                            }
                        }
                        entry.name == "build.gradle.kts" -> {
                            val gradleText = entry.inputStream().bufferedReader().use { it.readText() }
                            val match = Regex("""minSdkVersion\((\d+)\)""").find(gradleText)
                            if (match != null) {
                                minSdk = match.groupValues[1].toInt()
                            }
                        }
                        entry.name == "build.gradle.kts" -> {

                        }
                        entry.name.matches(Regex("classes\\.jar")) -> {
                            tempFile = makeTempFile("classes.jar").also {
                                entry.writeTo(it)
                            }
                        }
                    }
```

Reply:
Thanks for taking the time to write this out, but see my other comment for why I will regretfully decline this patch.